emukit/core/optimization/local_search_acquisition_optimizer.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from typing import List, Tuple
+
+import numpy as np
+
+from emukit.core import (CategoricalParameter, OneHotEncoding, OrdinalEncoding,
```

Comment:
All emukit internal imports should be relative

Reply:
Sure, i changed them all to relative imports.

Reply:
Imo core modules should only depend on other core modules. Currently both, random and local search, use the RandomDesign of the experimental_design module. Shall we create a optimizer module outside of core or move the random sampling to core, too?

Reply:
`emukit.optimization`? At the first glance that does not sound bad. @marpulli what do you think?

One tiny caveat is that we still need optimization interfaces in core. So one way would be to keep base class in core, and move the actual implementations to emukit.optimization

Reply:
The original idea was that non-core modules should rely on core only and not any other non-core module, right? `emukit.optimization` would rely on `emukit.experimental_design` which violates that idea. Do we still care? If we do then I guess the answer is to move random design to core.

Reply:
I kept all the optimizers in `core.optimization`. 

Several modules use the RandomDesign to generate random design. Therefore I decided to add a `sample_uniform` method directly to the parameter space to decouple them.